weblate/addons/models.py

```diff
@@ -110,13 +129,23 @@ def get_absolute_url(self):
         return reverse("addon-detail", kwargs={"pk": self.pk})
 
     def store_change(self, action):
-        Change.objects.create(
-            action=action,
-            user=self.component.acting_user,
-            component=self.component,
-            target=self.name,
-            details=self.configuration,
-        )
+        if self.component:
+            Change.objects.create(
+                action=action,
+                user=self.component.acting_user,
+                component=self.component,
+                target=self.name,
+                details=self.configuration,
+            )
+        elif self.project:
+            for component in self.project.component_set.all():
+                Change.objects.create(
+                    action=action,
+                    user=self.project.acting_user,
+                    component=component,
+                    target=self.name,
+                    details=self.configuration,
+                )
```

Comment:
Bind this to project instead:

```suggestion
                Change.objects.create(
                    action=action,
                    user=self.project.acting_user,
                    project=project,
                    target=self.name,
                    details=self.configuration,
                )
```